edx/frontend-app-admin-portal
# ENH: Move saml configuration from django admin to portal

Description:
This adds front end support for SAML Configuration such that customers can configure their third party authentication without direct intervention by CMS.

Notes:
- I implemented delete functionality before realizing we'd like soft deletes and that we're punting it. Instead of deleting deletes (heh) I left that functionality in here, but with a bit of a feature flag in the form of a state variable "deleteEnabled"
- Upon saving new providerData I am seeing a ReactState memory warning. Something something attempting to update state on unmounted component something something memory leak. If anyone can figure out what is actually causing that I'd love to know. The few state updates should be happening prior to the component unmounting AFAICT so not clear what is going on here.

Testing Configuration:
- You should probably just use my edx-platform branch from this PR: https://github.com/edx/edx-platform/pull/24557 otherwise errors are bound to happen 
- Enable the portal saml configuration screen on an enterprise customer.
  - http://localhost:18000/admin/enterprise/enterprisecustomer/
  - Click on a customer name
  - Scroll down and check the box next to "Enable portal saml configuration screen"
  - Click save at the bottom.
- Make sure you have a user with admin dashboard or operator role access *associated* with that enterprise.
- Then on a refresh of the portal, you should see the saml configuration icon in the sidebar of the frontend portal.
